### PR TITLE
Added a permission check for RNA related samples when opening a sample

### PR DESCRIPTION
**Pull request by ubuntolog**
Fixes https://github.com/imgag/ngs-bits/issues/343#event-7738153270

### CODE REVIEW

File: src/GSvar/MainWindow.cpp [hunk 1]
```diff
@@ -3671,6 +3671,8 @@ void MainWindow::loadFile(QString filename, bool show_only_error_issues)
 				// check for required files
 				foreach (const QString& rna_ps_id, db.getValues("SELECT id FROM processed_sample WHERE sample_id=:0", QString::number(rna_sample_id)))
 				{
+					if (!db.userCanAccess(LoginManager::userId(), rna_ps_id.toInt())) continue;
```
Comment (marc-sturm) [inline src/GSvar/MainWindow.cpp:3674]: The menu is enabled above. It should be disabled if there is no accessable RNA.

**Review by marc-sturm — CHANGES_REQUESTED** [review]

File: src/GSvar/MainWindow.cpp (hunk 1 above)
Comment (ubuntolog) [reply src/GSvar/MainWindow.cpp:3674]: I was not sure about the menu. Do you mean that all the items on the menu should be disabled? Even the one the screenshot?
image

Comment (marc-sturm) [reply src/GSvar/MainWindow.cpp:3674]: No, this entry should be active.

But then the line "rna_menu_btn_->setEnabled(true);" can be removed.
And all other entries should be disabled by default when GSvar starts.

Comment (ubuntolog) [reply src/GSvar/MainWindow.cpp:3675]: I have adjusted the menu code

**Review by marc-sturm — APPROVED** [review]

**Pull request merged**